doc/examples/applications/plot_cornea_spot_inpainting.py

```diff
@@ -0,0 +1,258 @@
+# Use local thresholding
+# ======================
+# To segment the dirt spots, we use thresholding. The images we are working
+# with have uneven illumination, which causes variations in the (absolute)
+# intensities of the foreground and the background, from one region to another
+# (distant) one. It is therefore more fitting to compute different threshold
+# values across the image, one for each region. This is called adaptive (or
+# local) thresholding, as opposed to the usual thresholding procedure which
+# employs a single (global) threshold for all pixels in the image.
+#
+# When calling the ``threshold_local`` function from the ``filters`` module,
+# we may change the default neighborhood size (``block_size``), i.e., the
+# typical size (number of pixels) over which illumination varies,
+# as well as the ``offset`` (shifting the neighborhood's weighted mean).
+# Let us try two different values for ``block_size``:
+
+thresh_1 = ski.filters.threshold_local(image_med, block_size=21, offset=15)
+thresh_2 = ski.filters.threshold_local(image_med, block_size=43, offset=15)
+
+mask_1 = image_med < thresh_1
+mask_2 = image_med < thresh_2
+
+#####################################################################
+# Let us define a convenience function to display two plots side by side, so
+# it is easier for us to compare them:
+
+def plot_comparison(plot1, plot2, title1, title2):
+    fig, (ax1, ax2) = plt.subplots(
+        ncols=2,
+        figsize=(12, 6),
+        sharex=True,
+        sharey=True
+    )
+    ax1.imshow(plot1, cmap='gray')
+    ax1.set_title(title1)
+    ax2.imshow(plot2, cmap='gray')
+    ax2.set_title(title2)
+
+plot_comparison(mask_1, mask_2, "block_size = 21", "block_size = 43")
+
+#####################################################################
+# The "dirt spots" appear to be more distinct in the second mask, i.e., the
+# one resulting from using the larger ``block_size`` value.
+# We noticed that increasing the value of the offset parameter from
+# its default zero value would yield a more uniform background,
+# letting the objects of interest stand out more visibly. Note that
+# toggling parameter values can give us a deeper
+# understanding of the method being used, which can typically move us
+# closer to the desired results.
+
+thresh_0 = ski.filters.threshold_local(image_med, block_size=43)
+
+mask_0 = image_med < thresh_0
+
+plot_comparison(mask_0, mask_2, "No offset", "offset = 15")
+
+#####################################################################
+# Remove fine-grained features
+# ============================
+# We use morphological filters to sharpen the mask and focus on the dirt
+# spots. The two fundamental morphological operators are *dilation* and
+# *erosion*, where dilation (resp. erosion) sets the pixel to the brightest
+# (resp. darkest) value of the neighborhood defined by a structuring element
+# (footprint).
+#
+# Here, we use the ``diamond`` function from the ``morphology`` module to
```

Comment:
Is there a specific reason to use `diamond` here and not `disk`? Disk seems more fitting to me because the mask is based on and equal Euclidian distance to the center...

Reply:
Good question and good point!

@ana42742 would you be able to rerun the tutorial with `disk` instead of `diamond` and check for possible differences in intermediate and final results? Thanks!

Reply:
@mkcor Sure! There is an observable difference in masks when using diamond v/s disk. As you can see, a disk footprint removes even more fine grained details than a diamond footprint.
<img width="965" alt="Screenshot 2023-06-29 at 1 53 02 AM" src="https://github.com/scikit-image/scikit-image/assets/95476298/d093e653-4808-42bd-b039-cd059a2333b2">

Reply:
@ana42742 thank you for trying out! It makes sense that more detail would be removed, since a disk covers a larger area wrt a diamond (of same size). Therefore, I would tend to stick with the diamond-shaped footprint, otherwise we don't catch enough spots. Does this sound good to you, @lagru?

Reply:
It does. Though, it brings up a point valid point: the footprint area. @ana42742 did you also decrease the size of the disk-shaped footprint while comparing? I'd be curious if a smaller `disk`-footprint with a roughly equivalent area would perform better.

Entirely optional though. I consider this thread resolved. ;)

Reply:
Included in cdb971d2aac8.